mgr/osd_support: new module for osd utility (#32677)
ceph/ceph: doc/mgr/osd_support.rst

OSD Support Module
==================
The OSD Support module holds osd specific functionality that
is needed by different components like the orchestrators.

In current scope:

> * osd draining

Reviewer: Maybe I'm bad a googling but I fail to find any description in the ceph docs as to what is `drain` is really about.
I guess that it is an attempt to tell the OSD to move its data to other OSDs without stalling PGs, but I guess it would not hurt to put that explicitly in the doc.

Author: `draining` probably isn't a well established term in the ceph universe.. yet :)

> I guess that it is an attempt to tell the OSD to move its data to other OSDs without stalling PGs, but I guess it would not hurt to put that explicitly in the doc.

Correct, it gracefully moves data from away from a specific OSD.

I'll add a sentence or two about `draining`

Author: done

OSD Support Module
==================
The OSD Support module holds osd specific functionality that
is needed by different components like the orchestrators.

In current scope:

* osd draining

Enabling
--------
When an orchestrator is used this should be enabled as a dependency.
(*currently only valid for the cephadm orchestrator*)

The *osd_support* module is manually enabled with::

  ceph mgr module enable osd_support

Commands
--------

Draining
########

This mode is for draining OSDs gracefully.

::

  ceph osd drain $osd_id

Takes a $osd_id and schedules it for draining. Since that process can take

> quite some time it will happen be put into the background. To query the status

Reviewer 2: ```suggestion
quite some time, the operation will be executed in the background. To query the status
```

of the operation you can:

::

  ceph osd drain status

This gives you the status of all running operation in this format::

  [{'osd_id': 0, 'pgs': 1234}, ..]

If you wish to stop an OSD from being drained::

  ceph osd drain stop [$osd_id]

> Stops all !SCHEDULED! osd drain operations (not the operations that have been started already)

Reviewer 2: ```suggestion
Stops all **scheduled** osd drain operations (not the operations that have been started already)
```
